src/app/models/AbstractModel.ts

```diff
@@ -138,3 +154,5 @@ export function getUnknownViewUrl(errorMsg: string) {
   console.warn(errorMsg);
   return unknownViewUrl;
 }
+
+export type ToJsonOptions = XOR<{ create: boolean }, { update: boolean }>;
```

Comment:
Rename to `toJSONOptions`?

Reply:
Since this is a type I think the adverb `To` aspect of this is the more incorrect aspect. The word `Json` is also overly specific.

Maybe `ApiPersistenceOptions` ?

Reply:
Upon understanding the context more: `ModelSerializationOptions`?

Reply:
It also seems like this type (although cool) is overly complex.

can't the parameter just be a string literal of `"create" | "update"`?

Reply:
It appears that only types have the ability to use `"create" | "update"`. Afaik, it wont be a feature until at least TS4.4: https://stackoverflow.com/a/65879197/1937822